File: 2230.py
import sys

n, m = map(int, sys.stdin.readline().split())

arr = []

for _ in range(n):
    arr.append(int(sys.stdin.readline()))

arr.sort()

l, r = 0, 0
ans = int(2000000000)

while l <= r and r < n:

    if arr[r] - arr[l] < m:
        r += 1

    else:
        ans = min(ans, arr[r] - arr[l])
        l += 1

print(ans)

File: 2230.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;


int main(){
    int m, n;
    cin >> n >> m;

    vector<int> a(n);

    for(int i=0; i<n; i++){
        cin >> a[i];
    }

    sort(a.begin(), a.end());
    
    int left = 0, right = 0;
    int ans = 2000000000;

    while (right < n){

        if (a[right]-a[left] >= m){
            ans = min(ans, a[right]-a[left]);
            left++;
        } 
        else{
            right++;
        }

        if(left == right) right++;
    }

    cout << ans ;

    return 0;
}

